src/dlstest/DlsClient.d

```diff
@@ -110,7 +99,7 @@ class DlsClient
         public void wait ( )
         {
             if ( this.pending > 0 )
-                this.task.suspend();
+                this.task_event.wait();
```

Comment:
I _think_ you don't need the `if ( this.pending > 0 )` now. The task event will only be triggered if `pending == 0` (line 141), and `TaskEvent.wait` only suspends the fiber if the event has not already been triggered (https://github.com/sociomantic-tsunami/ocean/blob/v3.x.x/src/ocean/task/util/Event.d#L55). Could you confirm this?

Reply:
I think your assumption is correct only for the case a request is done for one DLS node (or you are connected to only one DLS node) but it might not work for multiple nodes requests. Anyway I'll test it and confirm it.

Reply:
Well, it's difficult to try it with > 1 node, as the turtle env extension doesn't support that ;)

>I think your assumption is correct only for the case a request is done for one DLS node
What makes you think so?

Reply:
Sorry Gavin, I've got it wrong :s 
It is totally fine to remove the check `this.pending > 0` from here considering what you've stated about the way `TaskEvent.wait` works. (I've got confused for a similar check done in the notifier that should remain). I'll remove it the check from here.

> Well, it's difficult to try it with > 1 node, as the turtle env extension doesn't support that ;)

 I didn't know about that limitation. Thanks!